Remove flops warnings from the default profiler use case

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49896 Remove flops warnings from the default profiler use case**

Summary:
Add missing check for with_flops option set

Test Plan:
python test/test_profiler.py
CI

Benchmarking: https://gist.github.com/ilia-cher/5189904a08c4a12b24de1bf2ff9f1296

Differential Revision: [D25716930](https://our.internmc.facebook.com/intern/diff/D25716930)

Commit:
Remove flops warnings from the default profiler use case

Summary:
Add missing check for with_flops option set

Test Plan:
python test/test_profiler.py
CI

[ghstack-poisoned]

ngimel commented:
Can you please check if perf with no shapes and no flops is the same as it was pre- $46506?

xuzhao9 reviewed (approved):
LGTM

ilia-cher commented:
@ngimel , added benchmarking results (https://gist.github.com/ilia-cher/5189904a08c4a12b24de1bf2ff9f1296)

ilia-cher commented:
related to https://github.com/pytorch/pytorch/issues/49900 issue

Commit:
Update on "Remove flops warnings from the default profiler use case"


Summary:
Add missing check for with_flops option set

Test Plan:
python test/test_profiler.py
CI

Benchmarking: https://gist.github.com/ilia-cher/5189904a08c4a12b24de1bf2ff9f1296

Differential Revision: [D25716930](https://our.internmc.facebook.com/intern/diff/D25716930)

[ghstack-poisoned]

Commit:
Update on "Remove flops warnings from the default profiler use case"


Summary:
Add missing check for with_flops option set

Test Plan:
python test/test_profiler.py
CI

Benchmarking: https://gist.github.com/ilia-cher/5189904a08c4a12b24de1bf2ff9f1296

Differential Revision: [D25716930](https://our.internmc.facebook.com/intern/diff/D25716930)

[ghstack-poisoned]
